tdvt/tdvt/tdvt.py

```diff
@@ -621,40 +662,40 @@ def run_file(run_file, output_dir, threads, args):
 def main():
     parser, ds_registry, args = init()
 
-    if args.setup:
+    if args.command == 'action' and args.setup:
         print("Creating setup files...")
         create_test_environment()
         sys.exit(0)
-    if args.add_ds:
+    if args.command == 'action' and args.add_ds:
         add_datasource(args.add_ds, ds_registry)
         generate_files(ds_registry, True)
         sys.exit(0)
-    elif args.generate:
+    elif is_test(args) and args.generate:
         start_time = time.time()
         generate_files(ds_registry, True)
         end_time = time.time() - start_time
         print("Done: " + str(end_time))
-
         # It's ok to call generate and then run some tests, so don't exit here.
-    elif args.diff:
+    elif args.command == 'action' and args.diff:
         tdvt_invocation = TdvtInvocation(from_args=args)
         run_diff(tdvt_invocation, args.diff)
         sys.exit(0)
-    elif args.run_file:
-        output_dir = os.getcwd()
-        max_threads = get_level_of_parallelization(args)
-        sys.exit(run_file(args.run_file, output_dir, max_threads, args))
-    elif args.list_logical_configs is not None:
+    elif args.command == 'list' and args.list_logical_configs is not None:
         print_logical_configurations(ds_registry, args.list_logical_configs)
         sys.exit(0)
-    elif args.list_ds is not None:
-        print_configurations(ds_registry, args.list_ds, args.verbose)
+    elif args.command == 'list' and args.list_ds is not None:
+        print_configurations(ds_registry, [args.list_ds], args.verbose)
         sys.exit(0)
-
-    error_code = run_desired_tests(args, ds_registry)
-
-    sys.exit(error_code)
-
+    elif is_test(args):
+        if args.command == 'run' and args.run_file:
+            output_dir = os.getcwd()
+            max_threads = get_level_of_parallelization(args)
+            sys.exit(run_file(args.run_file, output_dir, max_threads, args))
+        error_code = run_desired_tests(args, ds_registry)
+        sys.exit(error_code)
+    
+    logging.error("Could not interpert arguments. Nothing done.")
+    sys.exit(-1)
```

Comment:
Would it make sense to do something like this here?

```suggestion
    else:
        logging.error("Could not interpret arguments. Nothing done.")
        p = create_parser()
        p.print_help()
        sys.exit(-1)
```
That would print the help text if the args are missing/weird, which might save the user a few keystrokes.

Reply:
Done